### DISCUSSION TITLE
Tree和TreeSelect的数据属性的命名问题

### DISCUSSION THREAD

**Discussion by fnoopv** [ant-design/ant-design#51447] [General]
这是`Tree`的
![image](https://github.com/user-attachments/assets/eae9c9aa-9387-49b6-9acd-0b20a2fb8369)
这是`TreeSelect`的
![image](https://github.com/user-attachments/assets/026d7f27-ab35-4ede-b3b2-f48044e69665)
值字段一个是`value`一个是`key`，想问一下是出于啥考虑的，同一份数据用在组件上要处理两次

**Comment by petercat-assistant** [comment]
在 Ant Design 中，`Tree` 和 `TreeSelect` 组件的属性命名不同，可能是出于以下几个考虑：

1. **组件功能差异**：`Tree` 和 `TreeSelect` 虽然都涉及树形结构的数据展示，但它们的功能和使用场景有所不同。`Tree` 主要用于展示层级结构，而 `TreeSelect` 则是一个选择器组件，允许用户从树形结构中选择值。因此，属性命名可能会根据组件的具体功能进行调整。

2. **历史原因**：在组件的开发过程中，可能由于历史原因导致了属性命名的差异。不同组件可能是在不同时间或由不同开发者开发的，导致了命名不一致。

3. **避免冲突**：在某些情况下，可能是为了避免与其他属性或方法名称冲突而选择不同的命名。

4. **用户体验**：可能是为了优化用户体验，使得每个组件的属性更符合其预期的使用方式。

如果你需要在两个组件之间共享数据，可以考虑在数据处理层进行一次性转换，以适应不同组件的需求。

如果你有更多关于这个问题的具体需求或建议，欢迎在 Ant Design 的 GitHub 讨论区提出。

如需进一步帮助，请在评论中描述您的问题并@petercat-assistant与我开始对话。

**Comment by Wxh16144** [comment]
> 最早 TreeSelect 组件和 Select 是分开研发，后来底层全部用 `rc-select` 改写了。但是历史债导致它的一些接口是没有统一的。

- see:  https://github.com/ant-design/ant-design/issues/51495#issuecomment-2456623533
